### PR TITLE
Physics Informed examples: Darcy Flow 

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Adds Darcy flow example using FNOs with two ways to introduce physics losses.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: examples/cfd/darcy_physics_informed/conf/config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+hydra:
+  job:
+    chdir: True
+  run:
+    dir: ./outputs_phy_deepo
+
+phy_wt: 0.1
```
Comment (mnabian) [inline examples/cfd/darcy_physics_informed/conf/config.yaml:21]: I suggest we add the configs here and not hard-code it in the training/inference code.

File: examples/cfd/darcy_physics_informed/conf/config_pino.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+hydra:
+  job:
+    chdir: True
+  run:
+    dir: ./outputs_pino
+
+phy_wt: 0.1
```
Comment (mnabian) [inline examples/cfd/darcy_physics_informed/conf/config_pino.yaml:21]: Same here

File: examples/cfd/darcy_physics_informed/download_data.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+filename = "datasets/Darcy_241/piececonst_r241_N1024_smooth1.hdf5"
+
+output_filenames = [
+    "datasets/Darcy_241/train.hdf5",
+    "datasets/Darcy_241/validation.hdf5",
+]
+
+# split_percentage = [80, 20]
+split_percentage = [10, 10]
```
Comment (mnabian) [inline examples/cfd/darcy_physics_informed/download_data.py:29]: Does this mean we are only using 20% of the data?

File: examples/cfd/darcy_physics_informed/test_ops.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import torch
```
Comment (mnabian) [inline examples/cfd/darcy_physics_informed/test_ops.py:1]: Is this script needed?

**Review by mnabian — APPROVED** [review]

File: examples/cfd/darcy_physics_informed/conf/config.yaml (hunk 1 above)
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/conf/config.yaml]: Done

File: examples/cfd/darcy_physics_informed/conf/config_pino.yaml (hunk 2 above)
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/conf/config_pino.yaml]: Done

File: examples/cfd/darcy_physics_informed/download_data.py (hunk 3 above)
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/download_data.py:29]: Yes, to simulate sparse data training.

File: examples/cfd/darcy_physics_informed/test_ops.py (hunk 4 above)
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/test_ops.py]: Good catch, removed.

**Comment by loliverhennigh** [comment]
This looks fantastic @ktangsali, I don't really see anything major to comment on as I think everything was in the older PR for this. I will approve

**Review by loliverhennigh — APPROVED** [review]

**Pull request merged**